external_parser/joiners/multistep_example_joiner.cc

```diff
@@ -115,12 +119,57 @@ void multistep_example_joiner::set_reward_function(const v2::RewardFunctionType
   }
 }
 
-void multistep_example_joiner::populate_order() {
-  //TODO: topological sort
+template<typename id_t, typename secondary_t, typename comp_t = std::greater<std::tuple<secondary_t, id_t>>>
```

Comment:
Add some comment about how to use this?
Just curious, why not use an existing library for topological sort?